### DISCUSSION TITLE
Sluggish performance

### DISCUSSION THREAD

**Discussion by maiertech** [community/community#26377] [Codespaces]
<p>This is the second week that I am noticing sluggish performance of my codespaces. VSCode remote explorer shows that my codespaces are running on a basic instance with 2 cores and 4 GB of RAM.</p>
<p>Before this started, my codespaces felt snappy and very responsive. Was there any change in what type of codespace instance is used by default for the beta? Is there a way to influence what type of instance is created?</p>

**Comment by 2percentsilk** [comment]
<p>Hi there!<br>
We haven’t made a change to change the instance types for the beta so that sluggishness is unexpected. Do you notice this more generally or is it in a specific part of your workflow - e.g. upon connection, creation, typing latency?</p>
<p>We’re looking to give you more control over the instance type but don’t have a timeline or more details to share on this work at the same time. We do know being able to select the power you need is important!</p>

**Comment by maiertech** [comment]
<p>It’s mostly typing in the terminal. I see a tiny delay before typed letters appear, but it’s not a huge thing probably also not completely avoidable.</p>

**Comment by jkeech** [comment]
<p>We made a switch yesterday to bump everyone up on the compute for new Codespaces (4 core, 8 GB RAM). We also have some storage perf improvements coming, but for terminal latency, neither of those changes are likely to help a lot.</p>
<p>Would you mind sharing your approximate location / country? We do plan to deploy Codespaces to more physical locations over time to bring them closer to users and reduce latency, although no ETA to share on that. I’m wondering what the current latency is from your location to our datacenters where Codespaces are running.</p>

**Comment by maiertech** [comment, marked as answer]
<p>Very cool! I did not pay attention to the VM specs today, but was wondering why everything feels much snappier :slight_smile:</p>
<p>My location is NYC but with a VPN to NYC in between. I think your VM upgrade might already have solved the issue. Will pay attention over the next few days. But let me know if you want me to test anything.</p>
